domain/goal/src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/goal/service/GoalService.kt

```diff
@@ -17,27 +17,38 @@ private val log = KotlinLogging.logger {}
  * [GoalPersistenceAdapter].
  *
  * This class is deliberately final so that it cannot be subclassed, ensuring that the business rules stay within the
- * domain.
+ * domain. Service method behaviour can however be customized and extended by creating an instance of this class with a
+ * [GoalServiceDecorator]. If not specified, the default implementation is a [NoopGoalServiceDecorator].
+ *
  */
 class GoalService(
   private val persistenceAdapter: GoalPersistenceAdapter,
+  private val goalServiceDecorator: GoalServiceDecorator = NoopGoalServiceDecorator(),
```

Comment:
It uses the no-op decorator by default, unless a different implementation is passed in on the constructor.